# Model Inference

## Import Libraries

In [2]:
# Libraries for Data Manipulation
import pandas as pd
# Libraries for saving
import pickle
import json

## Load Model

In [3]:
# Load model and other files
with open('model_rf.pkl', 'rb') as file_1:
  model_knn_inf = pickle.load(file_1)

# Load list kolom numerikal
with open('list_num_cols.txt', 'r') as file_2:
  num_cols_inf = json.load(file_2)

# Load list kolom kategorikal from JSON file
with open('list_cat_cols.txt', 'r') as file_3:
  cat_cols_inf = json.load(file_3)

In [22]:
# Data dibuat dengan random  dan dijadikan dataframe.

test = [{'BHK' : 2,
        'Size' : 1100, 
        'Floor' : 1, 
        'Area Type' : 'Super Area', 
        'Area Locality' : 'Bandel', 
        'City' : 'Kolkata',
        'Furnishing Status' : 'Unfurnished', 
        'Tenant Preferred' : 'Bachelors/Family', 
        'Bathroom' : 2,
        'Point of Contact' : 'Contact Owner'
}]

test_inference = pd.DataFrame(test)

# Tampilkan DataFrame
test_inference

BHK  Size  Floor   Area Type Area Locality     City Furnishing Status  \
0    2  1100      1  Super Area        Bandel  Kolkata       Unfurnished   

   Tenant Preferred  Bathroom Point of Contact  
0  Bachelors/Family         2    Contact Owner

In [23]:
# Transform the new data using the pipeline
predictions = model_knn_inf.predict(test_inference)

# Print the predictions
print("Predictions Harga Sewa:", predictions)


Predictions Harga Sewa: [14318.02132433]


**Insight** : Hasil prediksi menunjukkan bahwa harga sewa yang diprediksi oleh model Random Forest adalah sebesar 14318.02132433 . Dengan demikian, insight yang dapat diambil dari hasil ini adalah prediksi harga sewa untuk properti yang diberikan oleh model Random Forest.